lib/fog/google/requests/storage_json/get_bucket.rb

```diff
@@ -0,0 +1,88 @@
+module Fog
+  module Storage
+    class GoogleJSON
+      class Real
+        require "fog/google/parsers/storage/get_bucket"
+
+        # List information about objects in an Google Storage bucket
+        #
+        # ==== Parameters
+        # * bucket_name<~String> - name of bucket to list object keys from
+        # * options<~Hash> - config arguments for list.  Defaults to {}.
+        #   * 'ifMetagenerationMatch'<~Long> - Makes the return of the bucket metadata
+        #     conditional on whether the bucket's current metageneration matches the
+        #     given value.
+        #   * 'ifMetagenerationNotMatch'<~Long> - Makes the return of the bucket
+        #     metadata conditional on whether the bucket's current metageneration does
+        #     not match the given value.
+        #   * 'projection'<~String> - Set of properties to return. Defaults to 'noAcl',
+        #     also accepts 'full'.
+        #
+        # ==== Returns
+        # * response<~Excon::Response>:
+        #   * body<~Hash>:
+        #     See Google documentation on Bucket resource:
+        #     https://cloud.google.com/storage/docs/json_api/v1/buckets#resource
+        #
+
+        def get_bucket(bucket_name, options = {})
+          raise ArgumentError.new("bucket_name is required") unless bucket_name
+
+          api_method = @storage_json.buckets.get
+          parameters = {
+            "bucket" => bucket_name
+          }
+          parameters.merge! options
+
+          request(api_method, parameters)
+        end
+      end
+
+      class Mock
+        def get_bucket(bucket_name, options = {})
+          raise ArgumentError.new("bucket_name is required") unless bucket_name
+          response = Excon::Response.new
+          name = /(\w+\.?)*/.match(bucket_name)
+          if bucket_name == name.to_s
+            if bucket = data[:buckets][bucket_name]
```

Comment:
I know this is taken from the old API but no one should '=' in conditional without at least wrapping it.
Since this Mock looks a bit broken anyway, I would just wrap it.

Example from styleguide:
```
# bad (+ a warning)
if v = array.grep(/foo/)
  do_something(v)
  ...
end

# good (MRI would still complain though)
if (v = array.grep(/foo/))
  do_something(v)
  ...
end

# good
v = array.grep(/foo/)
if v
  do_something(v)
  ...
end
```